astropy/io/ascii/ui.py

```diff
@@ -579,39 +579,32 @@ def _guess(table, read_kwargs, format, fast_reader):
 
             reader.guessing = True
             dat = reader.read(table)
```

Comment:
It's either the `get_reader` or the `reader.read` that raises the exception. Whichever it is, lines before it can be moved before the `try` block and lines after to the `else` block.